src/Avalonia.Controls/SizeChangedEventArgs.cs

```diff
@@ -0,0 +1,88 @@
+﻿using Avalonia.Interactivity;
+using Avalonia.Layout;
+using Avalonia.Utilities;
+
+namespace Avalonia.Controls
+{
+    /// <summary>
+    /// Provides data specific to a SizeChanged event.
+    /// </summary>
+    public class SizeChangedEventArgs : RoutedEventArgs
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SizeChangedEventArgs"/> class.
+        /// </summary>
+        /// <param name="routedEvent">The routed event associated with these event args.</param>
+        public SizeChangedEventArgs(RoutedEvent? routedEvent)
+            : base (routedEvent)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SizeChangedEventArgs"/> class.
+        /// </summary>
+        /// <param name="routedEvent">The routed event associated with these event args.</param>
+        /// <param name="source">The source object that raised the routed event.</param>
+        public SizeChangedEventArgs(RoutedEvent? routedEvent, IInteractive? source)
+            : base(routedEvent, source)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SizeChangedEventArgs"/> class.
+        /// </summary>
+        /// <param name="routedEvent">The routed event associated with these event args.</param>
+        /// <param name="source">The source object that raised the routed event.</param>
+        /// <param name="previousSize">The previous size (or bounds) of the object.</param>
+        /// <param name="newSize">The new size (or bounds) of the object.</param>
+        public SizeChangedEventArgs(
+            RoutedEvent? routedEvent,
+            IInteractive? source,
+            Size previousSize,
+            Size newSize)
+            : base(routedEvent, source)
+        {
+            PreviousSize = previousSize;
+            NewSize = newSize;
+
+            // Only consider changed when the size difference is greater than LayoutEpsilon
+            // This compensates for any rounding or precision difference between layout cycles
+            HeightChanged = !MathUtilities.AreClose(newSize.Height, previousSize.Height, LayoutHelper.LayoutEpsilon);
+            WidthChanged = !MathUtilities.AreClose(newSize.Width, previousSize.Width, LayoutHelper.LayoutEpsilon);
```

Comment:
I think these properties can be implemented as a getter-methods. so it won't calculate these values for each event args created.

Reply:
Aside from that, looks good

Reply:
The calculation is trivial. I also don't want to change the API from WPF. I can do the following though:

`public bool HeightChanged => !MathUtilities.AreClose(newSize.Height...`

Reply:
I switched these to calculate on demand now and they are get-only properties.

 1. There is no caching so every time a getter is called it will re-calculate. I'm not worried about this.
 2. Is there any chance size could ever have `double.NaN` or infinity? My original assumption was no since bounds is actual size on the display... however, there will be a crash sometimes if that assumption is wrong.

Reply:
Thanks! Bounds with infinite size should cause an exception in layouting code before this event even raised. So should be safe.